ets-ecole-de-technologie-superieure.csl

```diff
@@ -163,21 +163,42 @@
           </group>
         </if>
         <else-if type="book" match="any">
-          <group delimiter=". " prefix=", ">
-            <text macro="edition"/>
-            <text macro="collection"/>
-            <group delimiter=", ">
-              <text macro="publisher"/>
-              <choose>
-                <if match="any" variable="number-of-pages">
-                  <text variable="number-of-pages" suffix=" p"/>
-                </if>
-                <else-if match="any" variable="page">
-                  <text variable="page" suffix=" p"/>
-                </else-if>
-              </choose>
-            </group>
-          </group>
+					<choose>
+            <if match="any" variable="edition">
+              <group delimiter=". " prefix=", ">
+                <text macro="edition"/>
+                <text macro="collection"/>
+                <group delimiter=", ">
+                  <text macro="publisher"/>
+                  <choose>
+                    <if match="any" variable="number-of-pages">
+                      <text variable="number-of-pages" suffix=" p"/>
+                    </if>
+                    <else-if match="any" variable="page">
+                      <text variable="page" suffix=" p"/>
+                    </else-if>
+                  </choose>
+                </group>
+              </group>
+            </if>
+            <else>
+              <group delimiter=". " prefix=". ">
+                <text macro="edition"/>
```

Comment:
@Bibref, why would you use the "edition" macro here, if the "edition" variable is empty here?